hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/monkies/PolicyBasedChaosMonkey.java

```diff
@@ -60,19 +66,30 @@ public PolicyBasedChaosMonkey(IntegrationTestingUtility util, Collection<Policy>
   }
 
   public PolicyBasedChaosMonkey(Properties monkeyProps, IntegrationTestingUtility util,
-    Policy... policies) {
-    this.monkeyProps = monkeyProps;
-    this.util = util;
-    this.policies = policies;
+    Collection<Policy> policies) {
+    this(monkeyProps, util, policies.toArray(new Policy[0]));
   }
 
   public PolicyBasedChaosMonkey(Properties monkeyProps, IntegrationTestingUtility util,
-    Collection<Policy> policies) {
+    Policy... policies) {
     this.monkeyProps = monkeyProps;
-    this.util = util;
-    this.policies = policies.toArray(new Policy[policies.size()]);
+    this.util = Objects.requireNonNull(util);
+    this.policies = Objects.requireNonNull(policies);
+    if (policies.length == 0) {
+      throw new IllegalArgumentException("policies may not be empty");
+    }
+    this.monkeyThreadPool = buildMonkeyThreadPool(policies.length);
   }
 
+  private static ExecutorService buildMonkeyThreadPool(final int size) {
+    return Executors.newFixedThreadPool(size, new ThreadFactoryBuilder()
```

Comment:
Would you prefer using our internal 
`Threads.newDaemonThreadFactory(String prefix, UncaughtExceptionHandler handler)` ?

Reply:
This one actually sets daemon to true, we don't want that so maybe we can have separate method for setting daemon to false. 
However, this work could be separate Jira and if we have majority, maybe we can make all Executor pool in HBase use our internal ThreadFactory and not from guava.

Reply:
It would be nice if we could either get away from using Guava helpers or get away from providing our own. Having both in the code keeps us guessing :)

Reply:
That's true, we should use only one of them, filed: https://issues.apache.org/jira/browse/HBASE-24750